### ISSUE TITLE
bug(bridge): Doctrine::setup()

### ISSUE THREAD

**Issue by MarcelWilhelmi** [Guikingone/SchedulerBundle#40]
Hello again ;-)

It seems like the auto_setup of the doctrine transport is not being executed. I used this as dsn in the scheduler.yaml.
`dsn: 'doctrine://default?execution_mode=first_in_first_out&table_name=scheduler&auto_setup=true'`

It looks like the Connection->create function of the doctrine connection class is always called first and this supposes, that the used database table already exists. But the setup() function will only be called in the executeQuery method.
The following error message is being displayed:
```
  An exception occurred while executing 'SELECT COUNT(t.id) as task_count FROM _symfony_scheduler_tasks t WHERE t.task_name = ? LOCK IN SHARE MODE' with params ["send-pending-messages"]:

  SQLSTATE[42S02]: Base table or view not found: 1146 Table 'bgr_web_dev._symfony_scheduler_tasks' doesn't exist
```

However I managed to call the setup function manually and the table _symfony_scheduler_tasks has been created with the columns id, task_name and body. Now when running `php bin/console scheduler:consume` I'm getting the following error:
```
 An exception occurred while executing 'INSERT INTO _symfony_scheduler_tasks (task_name, body) VALUES(?, ?) FOR UPDATE' with params ["send-pending-messages", "{\"body\":......"}"]:

  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'FOR UPDATE' at line 1
```

I guess INSERT INTO ... FOR UPDATE ... is not a valid statement, right? 

Kind regards,
Marcel

**Comment by Guikingone** [comment]
Ok, small update about this issue, I've found the main issue, by default, the `Connection` perform actions via a transaction (to prevent the use case where a task is accessed by 2 process at the same time), thing is, if the schema is not set outside of the transaction, the transaction still marked as active and we cannot call `setup()` which actually create the schema. 

A PR has been opened about the issue, it should be fixed pretty soon and released in `0.3.5`.

**Comment by Guikingone** [comment]
Hi @MarcelWilhelmi 👋🏻 

Small update about the fix, it should be ready pretty fast (I'm facing small issues with the tests as its almost a complete refactoring) and released in `0.4`, thanks again for the issue and sorry for the delay 🙂

**Comment by Guikingone** [comment]
Hi @MarcelWilhelmi 👋🏻 

The issue has been fixed via https://github.com/Guikingone/SchedulerBundle/pull/43, it should be released in `0.4` pretty soon.

**Comment by MarcelWilhelmi** [comment]
Thank you very much again :-) I will try it when version 0.4 has been released.
